modules/imgproc/src/connectedcomponents.cpp

```diff
@@ -1737,14 +1755,40 @@ int cv::connectedComponentsWithStats(InputArray _img, OutputArray _labels, Outpu
     _labels.create(img.size(), CV_MAT_DEPTH(ltype));
     cv::Mat labels = _labels.getMat();
     connectedcomponents::CCStatsOp sop(statsv, centroids);
-    if (ltype == CV_16U){
-        return connectedComponents_sub1(img, labels, connectivity, ccltype, sop);
+    int nbComponents;
+    if (ltype == CV_16U) {
+        nbComponents = connectedComponents_sub1(img, labels, connectivity, ccltype&(~CCL_FORE_BACK_GROUND), sop);
     }
-    else if (ltype == CV_32S){
-        return connectedComponents_sub1(img, labels, connectivity, ccltype, sop);
+    else if (ltype == CV_32S) {
+        nbComponents = connectedComponents_sub1(img, labels, connectivity, ccltype&(~CCL_FORE_BACK_GROUND), sop);
     }
-    else{
+    else {
         CV_Error(CV_StsUnsupportedFormat, "the type of labels must be 16u or 32s");
         return 0;
     }
+    if (!(ccltype & CCL_FORE_BACK_GROUND))
+        return nbComponents;
+    bitwise_not(img, img);
```

Comment:
Now function modifies the input image in `CCL_FORE_BACK_GROUND` mode, so sufficient note should be added to documentation.